library/ecp_curves.c

```diff
@@ -4661,6 +4668,7 @@ static int ecp_use_curve25519(mbedtls_ecp_group *grp)
 
     /* Actually, the required msb for private keys */
     grp->nbits = 254;
+    grp->h = 1;
```

Comment:
Cofactor to Curve 25519 is 8. To fix the segfault we need to update `mbedtls_ecp_group_free()` (We won't need to free `N` and `P` anymore).